src/nodes/limiter.rs

```diff
@@ -0,0 +1,381 @@
+    pub fn set_smooth_secs_down(&mut self, sample_rate: NonZeroU32, smooth_secs_down: f32) {
+        let coeff_down = SmoothingFilterCoeff::new(sample_rate, smooth_secs_down);
+        self.smooth_secs_down = smooth_secs_down;
+        self.coeff_down = coeff_down;
+    }
+
+    /// Return the next smoothed value.
+    #[inline(always)]
+    pub fn next_smoothed(&mut self) -> f32 {
+        if self.filter.z1 < self.target_value() {
+            self.filter
+                .process_sample_a(self.target_times_a_up, self.coeff_up.b1)
+        } else {
+            self.filter
+                .process_sample_a(self.target_times_a_down, self.coeff_down.b1)
+        }
+    }
+
+    /// Update the sample rate.
+    pub fn update_sample_rate(&mut self, sample_rate: NonZeroU32) {
+        self.coeff_up = SmoothingFilterCoeff::new(sample_rate, self.smooth_secs_up);
+        self.coeff_down = SmoothingFilterCoeff::new(sample_rate, self.smooth_secs_down);
+        self.target_times_a_up = self.target_value() * self.coeff_up.a0;
+        self.target_times_a_down = self.target_value() * self.coeff_down.a0;
+    }
+}
+
+/// Buffer.
+#[derive(Debug, Clone)]
+pub struct IncrementalMax {
+    // First item is unused for convenience. Buffer length is rounded up to an even number.
+    buffer: Box<[f32]>,
+    length: usize,
+    leaf_offset: usize,
+}
+
+impl IncrementalMax {
+    #[inline]
+    fn get_index(&self, i: usize) -> usize {
+        self.leaf_offset + i
+    }
+
+    /// Create a new [`IncrementalMax`].
+    pub fn new(length: usize) -> Self {
+        let leaf_offset = length.next_power_of_two();
+        Self {
+            buffer: vec![0.; leaf_offset + length + (length & 1)].into(),
+            length,
+            leaf_offset,
+        }
+    }
+
+    /// The length of the internal buffer.
+    #[inline]
+    // `is_empty` doesn't make sense for this type, the length should always be >0
+    #[allow(clippy::len_without_is_empty)]
+    pub fn len(&self) -> usize {
+        self.length
+    }
+
+    /// Get the maximum of the values in the buffer.
+    #[inline]
+    pub fn max(&self) -> f32 {
+        self.buffer[1]
+    }
+
+    /// Set a value at the given index.
+    pub fn set(&mut self, index: usize, value: f32) {
+        let mut i = self.get_index(index);
+
+        self.buffer[i] = value;
+
+        while i > 1 {
+            let max = self.buffer[i].max(self.buffer[i ^ 1]);
+            i >>= 1;
+            self.buffer[i] = max;
+        }
+    }
+
+    /// Clear the buffer, resetting all values to 0.
+    pub fn clear(&mut self) {
+        self.buffer.fill(0.);
+    }
+}
+
+/// Configuration for a [`LimiterNode`].
+#[derive(Debug, Clone, Component)]
+pub struct LimiterConfig {
+    /// The limiter lookahead - how much latency will be introduced in order to ensure that the
+    /// limiter will reduce volum in time for high peaks to be reduced. By default, it will set
+    /// the lookahead to the same as the `attack` of the limiter.
```

Comment:
Let's break up the docs here into two groups. Something like:

```rs
/// The limiter lookahead.
///
/// Lookahead determines how much latency will be introduced...
```

rustdoc will take the first line of an item's documentation and use it in concise contexts, like when listing all the items in a module. I don't think it matters for a field, but I prefer to use that same style everywhere.